feat(docs): READMEs for docs, Explanation and Reference Documentation (#148) (#168)
allinbits/cosmos-cash: docs/Reference/README.md

# Reference Documentation

How to use the Cosmos Cash Reference Documentation.

[TOC]

## Context

In Cosmos Cash we use the  *Grand Unified Theory of Documentation* (David Laing) as described by [Divio](https://documentation.divio.com/) as a basis for our documentation strategy.

This approach outlines four specific use cases for documentation:

* [Tutorials](../Tutorials/README.md)
* [How-Tos](../How-To/README.md)
* [Explanation](./Explanation/README.md)
* [Reference](./README.md)

This section addresses the last of these: **Reference documentation**. 

> For further background please see [the ADR relating to the documentation structure](./ADR/adr-002-docs-structure.md). 

Reviewer: see comment line 22 in docs/Explanation/README.md

hmmm I see a pattern here, do we really need to repeat the grand theory and core basics for each content type? wdyt about moving out and having a single instance of this information instead of repeating in each content type README?

## Approach

> As outlined in the strategy [Reference Documentation](https://documentation.divio.com/reference/) should be **information oriented**. It should allow the reader to easily navigate the content and use in conjunction with the code.

Reviewer: ```suggestion
As outlined in the strategy [Reference Documentation](https://documentation.divio.com/reference/) is **information-oriented**. Content must allow the reader to easily navigate the content and use the content in conjunction with the code.
```

This documentation describes the machinery, for example, classes, functions, interfaces, parameters, and so on.

## Contributing

> * The content should be dry, clear and terse in style.

Reviewer: ```suggestion
* The content must be dry, clear, and terse in style.
```

* All documentation should be written following [Google Documentation Best Practice](https://google.github.io/styleguide/docguide/best_practices.html)
* Generate as much documentation as possible from the code.
* Raise a PR for all documentation changes

## Layout

Reference Documentation will come in various forms:

> * [Architecture diagrams](./architecture) - this will be in SVG format so that it can be stored in GitHub and version controlled. 

Reviewer: ```suggestion
* [Architecture diagrams](./architecture) - Diagrams must be in SVG format so that the diagrams can remain crisp and clear at any resolution or size, stored in GitHub, and version controlled. 
```

Reviewer: we use png in our docs, so the SVG choice is probably based on: A vector image remains crisp and clear at any resolution or size. If so, let's add that reason

> * [Module specifications and designs](./MODULES.md) - By convention in Cosmos SDK module documentation is stored with the module itself. However we propose that easiest way for a new user to find documentation is at the root docs folder. To accomodate this module documentation will follow existing convention, but it will be reference from this section.

Reviewer: ```suggestion
* [Module specifications and designs](./MODULES.md) - By convention, module documentation in the Cosmos SDK is stored with the module itself. However, we propose that the easiest way for a new user to find documentation is to store the documentation at the root `docs` folder. To accomodate this module, the documentation will follow the existing convention, but the content will be reference from this section.
```

> * **Code-level documentation** - this will be text that is with the code and auto-generated from the code
> * **API reference** - this includes REST and gRPC enpoints.
> * [Glossary](,.GLOSSARY.md) - a dictionary of domain relevant terms. This can be used in conjunction with the [Cosmos Network Glossary](https://v1.cosmos.network/glossary).

Reviewer: ```suggestion
* **Code-level documentation** - The text that is part of the code and is used to auto-generate the documentation from the code.
* **API reference** - Including REST and gRPC endpoints.
* [Glossary](,.GLOSSARY.md) - a dictionary of domain-relevant terms. This glossary can be used in conjunction with the [Cosmos Network Glossary](https://v1.cosmos.network/glossary).
```
